src/object.jl

```diff
@@ -167,14 +167,20 @@ mutable struct PrefechedDownloadStream{T <: Object} <: IO
         )
         prefetch_size > 0 || throw(ArgumentError("`prefetch_size` must be positive, got $prefetch_size"))
         prefetch_multipart_size > 0 || throw(ArgumentError("`prefetch_multipart_size` must be positive, got $prefetch_multipart_size"))
-        for _ in 1:min(Threads.nthreads(), max(1, div(size, io.prefetch_multipart_size)))
-            Threads.@spawn _download_task($io)
+        if size > 0
+            for _ in 1:min(Threads.nthreads(), max(1, div(size, io.prefetch_multipart_size)))
```

Comment:
Like in the other PR, it seems like this should be the ceil()? If the size is 2.5 chunks you'd spawn 2 Tasks, but you probably want to spawn 3?

Reply:
also, it'd probably be good to split this computation into a function so that we can test it with a bunch of cases: 1 byte, exactly the chunk size, exactly 2 chunk sizes, more than nthreads chunks, etc.